### PR TITLE
Breakpoints, inline media queries, ampersand and pseudo presedence

### PR DESCRIPTION
**Pull request by christianalfoni**
Okay,

So this PR should contain:

1. Breakpoints as discussed here: https://github.com/modulz/stitches/issues/86#issuecomment-675590197
2. Ampersand support: https://github.com/modulz/stitches/issues/60
3. Pseudo presedence as mentioned here: https://github.com/modulz/stitches/issues/83#issuecomment-673942355

I wrote simple tests for these, but looks good so far 😄

### CODE REVIEW

**Review by peduarte — APPROVED** [review]
LGTM, but I'd like to run proper tests when its in master. 

@hadihallak can you review?

**Review by hadihallak — COMMENTED** [review]
Looks great 💯 
Thanks for implementing all of that against the updated master. very helpful.

This doesn’t implement the responsive atom warning logic we talked about in #86 right?

I think there might’ve been some confusion regarding breakpoints and media queries so apologies about that but the team agreed that mediaQueries is better suited than breakpoints even if we're gonna need to do some work internally separating breakpoints from nestable media queries.

@jjenzz also suggested a mixins idea which was very interesting and would solve the confusion we would have between breakpoints and media queries that we're likely to explore with later but until then the team agreed that mediaQueries provide a better experience so we're sticking with this.

In both cases, I think I could take it from here and implement this on top of this branch or if you’re still working here, I could do this in a separate branch as well.

Thanks again for putting this together 🙏

**Comment by jjenzz** [comment]
In the currently released version of stitches the following was possible but breaks with this PR https://codesandbox.io/s/kind-kalam-1itou

**Comment by hadihallak** [comment]
LIkely the product of #88 as it altered recession. will fix this and add a test for this use-case for the future

File: packages/css/src/index.ts [hunk 1]
```diff
@@ -125,14 +125,17 @@ const processStyleObject = (
     // shorthand css props or css props that has baked in handling:
     // see specificityProps in ./utils
     if (isSpecificityProp) {
-      const resolvedSpecificityProps = specificityProps[key](config.tokens, val);
+      const resolvedSpecificityProps = specificityProps[key](
+        config.tokens,
+        val
+      );
```
Comment (peduarte) [inline packages/css/src/index.ts:131]: prettier

File: packages/css/src/index.ts [hunk 2]
```diff
@@ -125,14 +125,17 @@ const processStyleObject = (
...
       processStyleObject(
         resolvedSpecificityProps,
         config,
         valueMiddleware,
         [...currentNestingPath],
         false,
-        false,
+        false
```
Comment (peduarte) [inline packages/css/src/index.ts:138]: prettier

**Review by hadihallak — APPROVED** [review]

**Comment by christianalfoni** [comment]
Hi there!

Sorry for late reply, a bit busy today 😄 

But this looks great! Happy it is in!!

**Pull request merged**